contracts/connectors/loantoken/LoanTokenBase.sol

```diff
@@ -74,4 +74,9 @@ contract LoanTokenBase is ReentrancyGuard, Ownable, Pausable {
 	/// The maximum trading/borrowing/lending limit per token address.
 	/// 0 -> no limit
 	mapping(address => uint256) public transactionLimit;
+
+	/// @notice A mapping of accounts stores those who
+	///   lend/withdraw on current block.
+	/// block => account => true/false
+	mapping(uint256 => mapping(address => bool)) internal hasInteracted;
```

Comment:
If I understand correctly, you can simplify it to `mapping(address => uint256) internal lastInteraction; // account => block ` 
and the check should be `lastInteraction[addr] < block.number`
it will save some gas, mostly due to storage reuse